mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -114,7 +114,12 @@ final class QuantileDiscretizer @Since("1.6.0") (@Since("1.6.0") override val ui
     splits(0) = Double.NegativeInfinity
     splits(splits.length - 1) = Double.PositiveInfinity
 
-    val bucketizer = new Bucketizer(uid).setSplits(splits)
+    val cutpoints = splits.distinct
+    if (splits.length != cutpoints.length) {
+      log.warn("Some quantiles were identical. Bucketing to " + (cutpoints.length - 1) +
```

Comment:
Nit: you can use string interpolation here. That part was fine. `s"... ${cutpoints.length - 1} ..."`. Super nit while we're still here: cutpoints is a term not used elsewhere. Just "distinctSplits"?